### ISSUE TITLE
Add JSON initialization methods

### ISSUE THREAD

**Issue by watzon** [crystal-data/num.cr#17]
It would be great to have the ability to convert a Tensor to JSON and vice versa. I know I can go the Tensor -> Array -> JSON route, but it should be pretty simple to add.

**Comment by christopherzimmerman** [comment]
There isn't even a good way to convert a Tensor to Array right now.  I need to work on some serializers, I'll probably get around to this when I actually implement reading/writing from files.

**Comment by christopherzimmerman** [comment]
Closed by #23

I handle the one dimensional case only, but you can pass a shape, so to store N-dimensional just store the shape separately.

```crystal
json_str = "[1, 2, 3]"

puts Tensor(Int32).from_json(json_str, [3])
# [1, 2, 3]

puts Tensor.random(0...10, [3]).to_json
# [7,6,2]
```
